pkg/api/v1alpha1/cluster.go

```diff
@@ -582,9 +608,22 @@ func validateGitOps(clusterConfig *Cluster) error {
 	gitOpsRefKind := gitOpsRef.Kind
 	fluxConfigActive := features.IsActive(features.GenericGitProviderSupport())
 
-	if gitOpsRefKind == FluxConfigKind && !fluxConfigActive {
-		return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
-			"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+	if gitOpsRefKind == FluxConfigKind {
+		if !fluxConfigActive {
+			return fmt.Errorf("FluxConfig and the generic git provider are not currently supported; " +
+				"to use this experimental feature, please set the environment variable GENERIC_GIT_PROVIDER_SUPPORT to true")
+		}
+		if fluxConfig.Spec.Git != nil {
+			if fluxConfig.Spec.Github != nil {
+				return errors.New("only one provider can be specified in FluxConfig. Please only specify either git or github")
+			}
+			if privateKeyFile, ok := os.LookupEnv(config.EksaGitPrivateKeyTokenEnv); !ok || len(privateKeyFile) <= 0 {
```

Comment:
Do we need any other validations on the values of these env vars?

Reply:
ideally we should do a check on the permissions of the private key file, it should be set at 600